cmd/mc/skin/add.go

```diff
@@ -0,0 +1,125 @@
+package skin
+
+import (
+	"errors"
+	"fmt"
+	"slices"
+
+	"github.com/google/uuid"
+	"github.com/mworzala/mc/internal/pkg/cli"
+	"github.com/spf13/cobra"
+)
+
+type addSkinOpts struct {
+	app *cli.App
+
+	account string
+	variant string
+	cape    string
+	name    string
+	apply   bool
+}
+
+var (
+	ErrInvalidType    = errors.New("invalid type")
+	ErrInvalidVariant = errors.New("invalid variant")
+
+	validVariants = []string{"classic", "slim"}
+)
+
+func newAddCmd(app *cli.App) *cobra.Command {
+	var o addSkinOpts
+
+	cmd := &cobra.Command{
+		Use:   "add",
+		Short: "Add a skin to your list",
+		Args: func(cmd *cobra.Command, args []string) error {
+			o.app = app
+			return o.validateArgs(cmd, args)
+		},
+		RunE: func(_ *cobra.Command, args []string) error {
+			o.app = app
+			return o.execute(args)
+		},
+	}
+
+	cmd.Flags().StringVar(&o.account, "account", "", "Account to use")
+	cmd.Flags().StringVar(&o.variant, "variant", "classic", "Skin variant [classic/slim]")
+	cmd.Flags().StringVar(&o.cape, "cape", "", "Cape name, 'none' to remove")
+	cmd.Flags().BoolVar(&o.apply, "apply", false, "Apply the skin")
+	cmd.Flags().BoolVar(&o.apply, "set", false, "Apply the skin")
```

Comment:
this makes me sad, but it seems cobra doesn't have a good option for arg aliases

Reply:
unfortunately it doesn't seem to, you also do the same thing in [modrinth/search.go](<https://github.com/mworzala/mc/blob/main/cmd/mc/modrinth/search.go#L47-L48>)